R/join_all.R

```diff
@@ -78,18 +82,51 @@ join_taxonomy2 <- function(austraits, vars =  c("family", "genus", "taxon_rank",
 #' @export
 #' @rdname join_all
 
-join_methods <- function(austraits, vars =  c("methods", "year_collected_start", "year_collected_end", "collection_type")) {
+join_methods <- function(austraits, ...) {
+  
+  # Switch for different versions
+  version <- what_version(austraits)
+  
+  if(what_version(austraits) %in% c("4-series", "5-series")){
+    version <- "new" 
+  } else
+    version <- "old"
+  
+  switch (version,
+          'new' = join_methods2(austraits, ...),
+          'old' = join_methods1(austraits, ...),
+  )
+}
+
+#' @title Joining methods info for AusTraits versions > 3.0.2
+#' @noRd
+#' @keywords internal
+join_methods2 <- function(austraits, vars =  c("methods", "year_collected_start", "year_collected_end", "collection_type")) {
+  austraits$methods %>% 
+    dplyr::select(c("dataset_id", "trait_name", "method_id"), tidyselect::any_of(vars)) %>% 
+    dplyr::distinct() -> methods
+  
+  austraits$traits <- austraits$traits %>%
+    dplyr::left_join(by=c("dataset_id", "trait_name", "method_id"),
+                     methods, relationship = "many-to-many")
+  
+  austraits
+}
+
+#' @title Joining methods info for AusTraits versions<== 3.0.2
+#' @noRd
+#' @keywords internal
+join_methods1 <- function(austraits, vars =  c("methods", "year_collected_start", "year_collected_end", "collection_type")) {
   austraits$methods %>% 
     dplyr::select(c("dataset_id", "trait_name"), tidyselect::any_of(vars)) %>% 
     dplyr::distinct() -> methods
   
   austraits$traits <- austraits$traits %>%
     dplyr::left_join(by=c("dataset_id", "trait_name"),
-                     methods)
+                     methods, relationship = "many-to-many")
```

Comment:
This seems very dangerous - is this to avoid warnings on an old AusTraits version that we already know duplicates lines? (I think GitHub mixed up as to what are actual "changes"

Reply:
This does seem dangerous! I don't recall adding this in. I've just had a test with and without relationship = many to many and doesn't make any difference so safer to remove for sure! Nice catch